libwget/http.c

```diff
@@ -29,7 +29,28 @@
  * RFC 6265
  *
  */
-
+#ifdef __cplusplus
```

Comment:
The same one more time! Move to an *internal place* IMHO.

Reply:
I am not 100% sure what you mean here.   I would assume I should move the assist macros to a common internal header we can just include, I assume the "private.h" potentially?